pkg/ipcache/kvstore.go

```diff
@@ -98,7 +99,7 @@ func newKVReferenceCounter(s store) *kvReferenceCounter {
 // UpsertIPToKVStore updates / inserts the provided IP->Identity mapping into the
 // kvstore, which will subsequently trigger an event in NewIPIdentityWatcher().
 func UpsertIPToKVStore(ctx context.Context, IP, hostIP netip.Addr, ID identity.NumericIdentity, key uint8,
-	metadata, k8sNamespace, k8sPodName string, npm types.NamedPortMap) error {
+	metadata, k8sNamespace, k8sPodName string, workload *models.Workload, npm types.NamedPortMap) error {
```

Comment:
Example here: No point to pass `k8snamespace` and `k8spodname` separately from workload, when workload includes these fields + kind. Maybe pass `*models.Workload` instead.